### ISSUE TITLE
unexpected error: Operator => should be surrounded by a single space

### ISSUE THREAD

**Issue by Mifrill** [bbatsov/rubocop#5063]
*optional*

I want to nicely align lines
```
  GLOSSARY = {
    en: {
      skipped_keys: /advertiser/i,
      subcategory:  /\AType\z/i,
      location:     /location/i,
      new:          /new/i,
      condition:    /Condition/i,
      yes:          "Yes",
      renames:      {
        "year of manufacture" => "year",
        "machine designation" => "stock_no"
      }
    },
    zh: {
      skipped_keys: /供货方状态/i,
      subcategory:  /\A类型\z/i,
      location:     /地点/i,
      new:          /新建/i,
      condition:    /状态/i,
      yes:          "是",
      renames:      {
        "制造年限"             => "year",
        "地点"                => "location",
        "经销商内部机器名称"    => "stock_no"
      }
    }
  }.freeze
```
Firstly: this is not a rubocop problem. It is impossible to align in the code editor (sublime or rubymine), we get this: https://imgur.com/a/B6lPW
this only happens with Chinese characters

Secondly: rubocop behaves strangely with these symbols
```

console: rubocop -a path_to_file.rb

path_to_file.rb :146:28: C: [Corrected] Operator => should be surrounded by a single space.
        "制造年限"             => "year",
                           ^^
path_to_file.rb:147:29: C: [Corrected] Operator => should be surrounded by a single space.
        "地点"                => "location",
                            ^^
path_to_file.rb:148:24: C: [Corrected] Operator => should be surrounded by a single space.
        "经销商内部机器名称"    => "stock_no"
                       ^^

1 file inspected, 3 offenses detected, 3 offenses corrected
```
tried the tabs and spaces, the result is the same

even one tab or space, provokes this error
```
C: [Corrected] Operator => should be surrounded by a single space.
        "经销商内部机器名称"  => "stock_no"
                     ^^
1 file inspected, 1 offense detected, 1 offense corrected
```

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.3 x86_64-linux)
```
I will be grateful for your help and thank you for developing

**Comment by Drenmi** [comment]
I can't really reproduce this. Using a single space between a string key and the operator causes no offense, regardless of the string content.

> even one tab or space, provokes this error
>
>```
> C: [Corrected] Operator => should be surrounded by a single space.
>         "经销商内部机器名称"  => "stock_no"
>                      ^^
> 1 file inspected, 1 offense detected, 1 offense corrected
>```

In this example, where you claim to use a single space, there is clearly two spaces. 🙂

**Comment by Mifrill** [comment]
@Drenmi  I mean one additional space or tab, i.e. If I add at least one space or tab, I get this error.

**Comment by Mifrill** [comment]
@Drenmi  I also do not have problems when one line, regardless of the content.
try to copy the hash completely and try
```
    add = {
      zh: {
        skipped_keys: /供货方状态/i,
        subcategory:  /\A类型\z/i,
        location:     /地点/i,
        new:          /新建/i,
        condition:    /状态/i,
        yes:          "是",
        renames:      {
          "制造年限" => "year",
          "地点" => "location",
          "经销商内部机器名称" => "stock_no"
        }
      }
    }
```

**Comment by Drenmi** [comment]
Then I am not sure what you are asking for in this issue. This is the exact thing that the cop is checking for. That there are no extra spaces around operators. 🙂  If you don't want this in your project, feel free to disable the cop in `.rubocop.yml`:

```yaml
# .rubocop.yml
Layout/SpaceAroundOperators:
  Enabled: false
```

**Comment by Mifrill** [comment]
## I will try to explain in parts

I have a hash:
```
  glossary = {
    en: {
      skipped_keys: /advertiser/i,
      subcategory: /\AType\z/i,
      location: /location/i,
      new: /new/i,
      condition: /Condition/i,
      yes: "Yes",
      renames: {
        "year of manufacture" => "year",
        "machine designation" => "stock_no"
      }
    },
    es: {
      skipped_keys: /Tipo del vendedor/i,
      subcategory: /\ATipo constructivo\z/i,
      location: /Localidad/i,
      new: /Nuevo/i,
      condition: /Estado/i,
      yes: "Sí",
      renames: {
        "año de fabricación" => "year",
        "localidad" => "location",
        "denominación de máquina del distribuidor" => "stock_no"
      }
    }
  }
```
check
```
# console: rubocop -a filename.rb
1 file inspected, no offenses detected
```
## *I want to align all the lines* to look better
```
  glossary = {
    en: {
      skipped_keys: /advertiser/i,
      subcategory:  /\AType\z/i,
      location:     /location/i,
      new:          /new/i,
      condition:    /Condition/i,
      yes:          "Yes",
      renames:      {
        "year of manufacture"                      => "year",
        "machine designation"                      => "stock_no"
      }
    },
    es: {
      skipped_keys: /Tipo del vendedor/i,
      subcategory:  /\ATipo constructivo\z/i,
      location:     /Localidad/i,
      new:          /Nuevo/i,
      condition:    /Estado/i,
      yes:          "Sí",
      renames:      {
        "año de fabricación"                       => "year",
        "localidad"                                => "location",
        "denominación de máquina del distribuidor" => "stock_no"
      }
    }
  }
```
check
```
# console: rubocop -a filename.rb
1 file inspected, no offenses detected
```
## now I'm adding a Chinese locale
```
  glossary = {
    en: {
      skipped_keys: /advertiser/i,
      subcategory:  /\AType\z/i,
      location:     /location/i,
      new:          /new/i,
      condition:    /Condition/i,
      yes:          "Yes",
      renames:      {
        "year of manufacture"                      => "year",
        "machine designation"                      => "stock_no"
      }
    },
    es: {
      skipped_keys: /Tipo del vendedor/i,
      subcategory:  /\ATipo constructivo\z/i,
      location:     /Localidad/i,
      new:          /Nuevo/i,
      condition:    /Estado/i,
      yes:          "Sí",
      renames:      {
        "año de fabricación"                       => "year",
        "localidad"                                => "location",
        "denominación de máquina del distribuidor" => "stock_no"
      }
    },
    zh: {
      skipped_keys: /供货方状态/i,
      subcategory:  /\A类型\z/i,
      location:     /地点/i,
      new:          /新建/i,
      condition:    /状态/i,
      yes:          "是",
      renames:      {
        "制造年限" => "year",
        "地点" => "location",
        "经销商内部机器名称" => "stock_no"
      }
    }
  }
```
check
```
# console: rubocop -a filename.rb
1 file inspected, no offenses detected
```
## Now I want to align the lines of the `rename` key just like en and es locales
```
  glossary = {
    en: {
      skipped_keys: /advertiser/i,
      subcategory:  /\AType\z/i,
      location:     /location/i,
      new:          /new/i,
      condition:    /Condition/i,
      yes:          "Yes",
      renames:      {
        "year of manufacture"                      => "year",
        "machine designation"                      => "stock_no"
      }
    },
    es: {
      skipped_keys: /Tipo del vendedor/i,
      subcategory:  /\ATipo constructivo\z/i,
      location:     /Localidad/i,
      new:          /Nuevo/i,
      condition:    /Estado/i,
      yes:          "Sí",
      renames:      {
        "año de fabricación"                       => "year",
        "localidad"                                => "location",
        "denominación de máquina del distribuidor" => "stock_no"
      }
    },
    zh: {
      skipped_keys: /供货方状态/i,
      subcategory:  /\A类型\z/i,
      location:     /地点/i,
      new:          /新建/i,
      condition:    /状态/i,
      yes:          "是",
      renames:      {
        "制造年限"                                 => "year",
        "地点"                                     => "location",
        "经销商内部机器名称"                        => "stock_no"
      }
    }
  }
```
## and for some reason I can not do this
check
```
# console: rubocop -a filename.rb
filename.rb:40:48: C: [Corrected] Operator => should be surrounded by a single space.
        "制造年限"                                 => "year",
                                               ^^
filename.rb:41:50: C: [Corrected] Operator => should be surrounded by a single space.
        "地点"                                     => "location",
                                                 ^^
filename.rb:42:44: C: [Corrected] Operator => should be surrounded by a single space.
        "经销商内部机器名称"                        => "stock_no"
                                           ^^

1 file inspected, 3 offenses detected, 3 offenses corrected
```
why there is no error, in other locales, except for Chinese?

**Comment by Mifrill** [comment]
## remove other locales, leave only the Chinese, and try to align the key line renames
```
  glossary = {
    zh: {
      skipped_keys: /供货方状态/i,
      subcategory:  /\A类型\z/i,
      location:     /地点/i,
      new:          /新建/i,
      condition:    /状态/i,
      yes:          "是",
      renames:      {
        "制造年限"          => "year",
        "地点"             => "location",
        "经销商内部机器名称" => "stock_no"
      }
    }
  }
```
for some reason unknown to me, it is impossible to align the lines, the symbol "=>" is still disjointed with respect to the neighboring same symbols, and it is impossible to align it. Try it yourself

check
```
# console: rubocop -a filename.rb
 filename.rb:15:25: C: [Corrected] Operator => should be surrounded by a single space.
        "制造年限"          => "year",
                        ^^
 filename.rb:16:26: C: [Corrected] Operator => should be surrounded by a single space.
        "地点"             => "location",
                         ^^

1 file inspected, 2 offenses detected, 2 offenses corrected
```
why is this happening?

**Comment by EhsanZ** [comment]
How these lines look like after getting corrected? I tried to reproduce your issue but Rubocop wouldn't detect any offense.

**Comment by Mifrill** [comment]
@EhsanZ 
OS
```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 17.10
Release:	17.10
Codename:	artful
```
after getting corrected:
```
    zh: {
      skipped_keys: /供货方状态/i,
      subcategory:  /\A类型\z/i,
      location:     /地点/i,
      new:          /新建/i,
      condition:    /状态/i,
      yes:          "是",
      renames:      {
        "制造年限" => "year",
        "地点" => "location",
        "经销商内部机器名称" => "stock_no"
      }
    }
```

**Comment by Drenmi** [comment]
Right. This cop will allow multiple spaces if they are used to align operators on successive lines. That is, if the operator appears in the same column as one on the previous or the following line.

In the case of the Spanish hash, the operators are in the same column. In the case of the Chinese, they are not. It may look like it, because of the varying width of the characters, but they are actually in columns: 22, 23, and 18.

**Comment by Mifrill** [comment]
Obviously this is not a rubocop problem, the alignment is related to the specifics of Asian characters

**Comment by Mifrill** [comment]
```
  def renames
    {
      "รุ่นเครื่องจักร"      => "model",
      "หมายเลขเครื่องจักร" => "stock_no",
      "ปีที่ผลิต"           => "year"
    }
  end
```
rubocop -a
```
path_to_file: C: [Corrected] Operator => should be surrounded by a single space.
      "รุ่นเครื่องจักร"      => "model",
                             ^^
path_to_file: C: [Corrected] Operator => should be surrounded by a single space.
      "ปีที่ผลิต"           => "year"
                            ^^

1 file inspected, 2 offenses detected, 2 offenses corrected
```
after correcting
```
  def renames
    {
      "รุ่นเครื่องจักร" => "model",
      "หมายเลขเครื่องจักร" => "stock_no",
      "ปีที่ผลิต" => "year"
    }
  end
```
it seems that with the Thai symbols the same thing happens
